### DISCUSSION TITLE
How to keep climbing for F-15

### DISCUSSION THREAD

**Discussion by oliver29520** [JSBSim-Team/jsbsim#461] [Q&A]
I tuned 'fcs/elevator-cmd-norm' and 'fcs/throttle-cmd-norm' to climb for F-15.
but it is only able to climb 1000 ft even though I set 'fcs/throttle-cmd-norm' to 1.0.
Why does it not work and cannot keep outputing the power?

**Comment by seanmcleod** [comment]
You haven't provided enough information regarding your initial conditions, or any data/graphs showing the output you're seeing in order for someone to help diagnose. 

So what I've knocked together is the following example using the python interface, to basically have the F-15 model trimmed at 10,000ft at 350KIAS with a flight path angle (gamma) of 5 degrees, and then I've plotted it's altitude for the next 120s.

You can also do the equivalent using an initial conditions file and/or a script file if you prefer. Not sure how you're 'driving' JSBSim in your case.

```python
import jsbsim
import matplotlib.pyplot as plt

# The path supplied to FGFDMExec is the location of the folders "aircraft", "engines" and "systems"
fdm = jsbsim.FGFDMExec('..\\') 

# Load the aircraft 
fdm.load_model('f15') 

# Set engines running
fdm['propulsion/engine[0]/set-running'] = 1
fdm['propulsion/engine[1]/set-running'] = 1

results = []

fdm['ic/h-sl-ft'] = 10000
fdm['ic/vc-kts'] = 350
fdm['ic/gamma-deg'] = 5

# Initialize the aircraft with initial conditions
fdm.run_ic() 
fdm.run()

# Trim
try:
    fdm['simulation/do_simple_trim'] = 1

    while fdm.run() and fdm.get_sim_time() <= 120:
        mach = fdm['velocities/mach']
        results.append((fdm.get_sim_time(), fdm['position/h-sl-ft']))

except RuntimeError as e:
    # The trim cannot succeed. Just make sure that the raised exception
    # is due to the trim failure otherwise rethrow.
    if e.args[0] != 'Trim Failed':
        raise

time, altitude = zip(*results)
plt.plot(time, altitude)

plt.xlabel('Time (s)')
plt.ylabel('Altitude (ft)')
plt.title('Altitude vs Time')

plt.show()
```

So first off you'll see the following trim solution printed out. So 70% throttle and pitch trim of -0.05.

```sh
  Full Trim

  Trim successful
  Trim Results:
       Angle of Attack:   1.71  wdot:  9.62e-05 Tolerance: 1.000000e-03  Passed
              Throttle:   0.70  udot: -2.39e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.05  qdot:  2.44e-10 Tolerance: 1.000000e-04  Passed
            Roll Angle:   0.02  vdot:  1.21e-07 Tolerance: 1.000000e-03  Passed
              Ailerons:   0.00  pdot:  4.17e-09 Tolerance: 1.000000e-04  Passed
                Rudder:   0.00  rdot: -9.27e-22 Tolerance: 1.000000e-04  Passed
```

With the following altitude versus time plot.

![image](https://user-images.githubusercontent.com/824856/123830017-38712300-d903-11eb-94cb-f957f6348ffd.png)

**Reply by oliver29520** [reply]
Thanks for your answer.
I find that the F-15 is multi-engine and has multiple throttle_cmd and mixture_cmd controls. It's work when I set throttle-cmd-norm, mixture-cmd-norm, throttle-cmd-norm[1] and mixture-cmd-norm[1]. However, I still do not know how to use mixture-cmd-norm or mixture-cmd-norm[1].  Do you know that? or I can find it somewhere.

**Reply by theorynice** [reply]
> Thanks for your answer. I find that the F-15 is multi-engine and has multiple throttle_cmd and mixture_cmd controls. It's work when I set throttle-cmd-norm, mixture-cmd-norm, throttle-cmd-norm[1] and mixture-cmd-norm[1]. However, I still do not know how to use mixture-cmd-norm or mixture-cmd-norm[1]. Do you know that? or I can find it somewhere.

Do you mean only set throttle-cmd-norm cannot control the F-15's velocity?

**Comment by seanmcleod** [comment]
The `mixture_cmd` control is only relevant for piston engines, not for turbine engines, so you can ignore it for the F-15.
